proposals/0207-rpc-message-protection.md

```diff
@@ -171,51 +207,17 @@ Multiple function groups can include the same RPC; each group has its own flag f
-
-```xml
-<function name="OnPermissionsChange" functionID="OnPermissionsChangeID" messagetype="notification" since="2.0">
-    <description>Provides update to app of which policy-table-enabled functions are available</description>
-    <param name="permissionItem" type="PermissionItem" minsize="0" maxsize="500" array="true" mandatory="true">
-        <description>Change in permissions for a given set of RPCs</description>
-    </param>
-+   <param name="needProtection" type="Boolean"  mandatory="false" since="5.1"/>
-</function>
-```
-
-```json
-"app_policies": {
-    "default": {
-+     "need_protection": false,
-      "keep_context": false,
-      "steal_focus": false,
-      "priority": "NONE",
-      "default_hmi": "NONE",
-      "groups": [
-        "Base-4"
-      ]
-    },
-    "appid_123456789": {
-+     "need_protection": true,
-      "keep_context": false,
-      "steal_focus": true,
-      "priority": "NONE",
-      "default_hmi": "NONE",
-      "groups": [
-        "Base-4", "RemoteControl"
-      ]
-    },
-    ...
-}
-```
-
-In this alternative solution, OEMs are forced to choose either no encryption at all or almost all messages get encrypted. OEMs lose the flexibility to configure which function group of RPCs need encryption. Not all OEMs are ready to encrypt all RPC messages for a variety of reasons. At the same time, there is a need to encrypt remote control related RPCs that have a high risk for the new apps on the existing hardware. This proposal proposes the solution of encrypting select RPCs rather than encrypting all RPCs.
 
 ### SDL server updates:
-The SDL server may support the new parameter in the policy table. 
+
+The SDL server need to support the new parameters in the policy table.
+* Database and API modifications to support the new `encryption_required` attribute of Functional Groups.
+* UI addition of a checkbox or toggle-switch to enable/disable the state of `encryption_required` for a Functional Group.
+* Database and API modifications to support the new `encryption_required` app_policy attribute.
+* New `encryption_required` server environment variable to indicate whether auto-approved applications should automatically have their `encryption_required` app_policy value set to `true` or `false` (default `false`)
+* UI addition of a checkbox or toggle-switch to enable/disable an individual application version's `encryption_required` app_policy value.
```

Comment:
Update to:
UI addition of a checkbox or toggle-switch to enable/disable an individual application **version** `encryption_required` app_policy value.

Reply:
updated